### ISSUE TITLE
sambamba installation error: 

### ISSUE THREAD

**Issue by asinode** [lomereiter/sambamba#200]
I just cloned and tried to install `sambamba` with `git clone --recursive` option.

After I ran `make sambamba-ldmd2-64`, the installation stopped with the following error.

```
cd lz4/lib && cc -O3 -c lz4.c lz4hc.c lz4frame.c xxhash.c && ar rcs liblz4.a lz4.o lz4hc.o lz4frame.o xxhash.o
mkdir -p build/
ldmd2 @sambamba-ldmd-release.rsp
make: ldmd2: Command not found
make: *** [sambamba-ldmd2-64] Error 127
```

Pls help! Thanks.
~ Anand

**Comment by lomereiter** [comment]
Hi, you need to install LDC compiler first: https://github.com/ldc-developers/ldc/releases/tag/v0.16.1

**Comment by asinode** [comment]
@lomereiter thank you.

I did what you said. I have a new set of issues now. Sorry I am not so good with linux.

I installed LDC compiler from the link you gave. And then, I extracted it and tried to create a symlink to be found in the executable search path (as per the README in the LDC folder). 

Here are the steps I followed.

1. After extracting LDC tar file, I retried building `sambamba` and I got a new error.
```
$ make sambamba-ldmd2-64 
cd htslib && make
make[1]: Entering directory `/home/suranand/sambamba/htslib'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/suranand/sambamba/htslib'
mkdir -p build/
ldmd2 @sambamba-ldmd-release.rsp
/home/suranand/ldc2-0.16.1-linux-x86_64/bin/../import/std/numeric.d(1606): Deprecation: read-modify-write operations are not allowed for shared variables. Use core.atomic.atomicOp!"+="(s, e) instead.
/home/suranand/ldc2-0.16.1-linux-x86_64/bin/../import/std/numeric.d(1632): Deprecation: read-modify-write operations are not allowed for shared variables. Use core.atomic.atomicOp!"*="(e, f) instead.
gcc -Wl,--gc-sections -o build/sambamba build/sambamba.o -Lhtslib -Llz4/lib -Wl,-Bstatic -lhts -llz4 -Wl,-Bdynamic -l:libphobos2-ldc.a -l:libdruntime-ldc.a  -lrt -lpthread -lm
/usr/bin/ld: cannot find libphobos2-ldc.a
/usr/bin/ld: cannot find libdruntime-ldc.a
collect2: error: ld returned 1 exit status
make: *** [sambamba-ldmd2-64] Error 1
```

2. Since it shows the `/usr/bin/ld`, I tried to create a symlink using the following command. But it seems, there already is symlink called `ld`. 

```
$ ln -s /home/suranand/ldc2-0.16.1-linux-x86_64/lib /usr/bin/ld
ln: failed to create symbolic link ‘/usr/bin/ld’: File exists
```

3. So I tried to see what the existing `ld` symlink is.

```
$ file /usr/bin/ld
/usr/bin/ld: symbolic link to `ld.bfd' 
suranand@asinode:~/sambamba$ locate ld.bfd
/usr/bin/ld.bfd
/usr/bin/x86_64-linux-gnu-ld.bfd
/usr/share/man/man1/ld.bfd.1.gz
/usr/share/man/man1/x86_64-linux-gnu-ld.bfd.1.gz
```

4. I tried to change the symlink name as `ldc` and wanted to see if `sambamba` could locate it. But I still got the same error.

```
/usr/bin/ld: cannot find libphobos2-ldc.a
/usr/bin/ld: cannot find libdruntime-ldc.a
collect2: error: ld returned 1 exit status
make: *** [sambamba-ldmd2-64] Error 1
```

Hope this info helps you in finding a solution for me. Thanks.

**Comment by lomereiter** [comment]
Sorry, I forgot to mention that you should also set `LIBRARY_PATH` env. variable:
```bash
export LIBRARY_PATH=<path to ldc directory>/lib
```

There is no need to tinker with `/usr/bin/ld` or any other system-installed tools.

**Comment by ezjung** [comment]
I also have same issue. I exported LIBRARY_PATH to ldc directory, but still sambamba installation was successful. I am using Ubuntu 14.04 and ldc 0.17.1 version. ldc readme file wasn't that helpful to install ldc. Can you explain more explicitly? Thanks in advance for your solution and also nice tool.

**Comment by ezjung** [comment]
My error message is ;

sp@sp-ThinkPad-X230:~/Tools/sambamba$ make
cd htslib && make
make[1]: Entering directory `/home/sp/Tools/sambamba/htslib'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/sp/Tools/sambamba/htslib'
mkdir -p build/
rdmd --force --build-only --compiler=dmd -IBioD -g -d -L-Lhtslib -L-l:libhts.a -L-l:libphobos2.a -L-Llz4/lib -L-l:liblz4.a -ofbuild/sambamba main.d
make: rdmd: Command not found
make: *** [all] Error 127

Installation seems require "rdmd". What should I do?

**Comment by lomereiter** [comment]
@ezjung you should run `make sambamba-ldmd2-64`; default `make` target is the development one, compiled with DMD instead of LDC.

**Comment by ezjung** [comment]
Compiling went through, but linux could not find "sambamba" command. While installation process, I've got message below. I do not thinks it is properly installed. Can you please check the message below and give me some suggestion? Thanks.

sp@sp-ThinkPad-X230:~/Tools/sambamba$ make sambamba-ldmd2-64
cd htslib && make
make[1]: Entering directory `/home/sp/Tools/sambamba/htslib'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/sp/Tools/sambamba/htslib'
mkdir -p build/
ldmd2 @sambamba-ldmd-release.rsp
/home/sp/Tools/ldc2-0.17.1-linux-x86_64/bin/../import/std/numeric.d(1616): Deprecation: read-modify-write operations are not allowed for shared variables. Use core.atomic.atomicOp!"+="(s, e) instead.
/home/sp/Tools/ldc2-0.17.1-linux-x86_64/bin/../import/std/numeric.d(1642): Deprecation: read-modify-write operations are not allowed for shared variables. Use core.atomic.atomicOp!"*="(e, f) instead.
gcc -Wl,--gc-sections -o build/sambamba build/sambamba.o -Lhtslib -Llz4/lib -Wl,-Bstatic -lhts -llz4 -Wl,-Bdynamic -l:libphobos2-ldc.a -l:libdruntime-ldc.a  -lrt -lpthread -lm

**Comment by lomereiter** [comment]
Looks good, just some compiler warning messages. The executable should be in `build/` subdirectory, you can either add the subdir to `PATH` or just copy the tool into `/usr/local/bin`.

**Comment by ezjung** [comment]
Thanks a LOT! 

It's working. D-compiler is real headache for linux novice. 

Now I need to do real work using your program. Thanks again!

**Comment by vd4mmind** [comment]
I installed the LDC 
```
echo $PATH
/home/vdas/tools/ldc2-0.17.1-linux-x86_64/bin
echo $LD_LIBRARY_PATH
/home/vdas/tools/ldc2-0.17.1-linux-x86_64/lib:

```
Then I did the following in /home/vdas/tools directory

```
git clone --recursive https://github.com/lomereiter/sambamba.git
    cd sambamba
    make sambamba-ldmd2-64
```
Installation was in progress but I got the following error which means that there is an error in the make.
```
ldmd2 @sambamba-ldmd-release.rsp
/home/vdas/tools/ldc2-0.17.1-linux-x86_64/bin/../import/std/numeric.d(1616): Deprecation: read-modify-write operations are not allowed for shared variables. Use core.atomic.atomicOp!"+="(s, e) instead.
/home/vdas/tools/ldc2-0.17.1-linux-x86_64/bin/../import/std/numeric.d(1642): Deprecation: read-modify-write operations are not allowed for shared variables. Use core.atomic.atomicOp!"*="(e, f) instead.
gcc -Wl,--gc-sections -o build/sambamba build/sambamba.o -Lhtslib -Llz4/lib -Wl,-Bstatic -lhts -llz4 -Wl,-Bdynamic -l:libphobos2-ldc.a -l:libdruntime-ldc.a  -lrt -lpthread -lm
/usr/bin/ld: cannot find libphobos2-ldc.a
/usr/bin/ld: cannot find libdruntime-ldc.a
collect2: error: ld returned 1 exit status
make: *** [sambamba-ldmd2-64] Error 1
```
How do I get rid of this error to properly install the sambamba and then put the path in my `bashrc` so that I can run it for my automatic workflows. Thanks

**Comment by pjohansen** [comment]
Hi I am trying to install sambamba on ou red hat linux server
I have downloaded and unpacked ldc and added to path, i can call ldc2 and ldmd2 commands, yet the sambamba installation cannot locate the ldc2 executable. Is there a way other than path to tell sambamba where to find ldc2. Any help would be appreciated

[pjoh0061@rghkromesx001 sambamba]$ sudo make sambamba-ldmd2-64
cd htslib && make
make[1]: Entering directory `/home/pjoh0061/sambamba/htslib'
make[1]: Nothing to be done for `all'.
make[1]: Leaving directory `/home/pjoh0061/sambamba/htslib'
mkdir -p build/
ldmd2 @sambamba-ldmd-release.rsp
Error: Could not locate ldc2 executable.
make: *** [sambamba-ldmd2-64] Error 1

**Comment by sambrightman** [comment]
Both steps are mentioned in the discussion  above:

```
export PATH=$PATH:/path/to/ldc/bin
export LIBRARY_PATH=/path/to/ldc/lib
```

**Comment by lomereiter** [comment]
Hi @pjohansen, I presume that's because you're running `make` using `sudo`. As a sidenote, it's not recommended to run `make` under root in general.

**Comment by pjohansen** [comment]
Thank you, i ran sudo because i keep getting permission denied. i will look at the permissions.
Thank you for a quick reply

**Comment by AndreaHobby** [comment]
I followed the instructions above and I still can't run sambamba. 
1. I cloned the sambamba repo from GitHub. 
2. I used the sudo apt get install command to install the ldc. I am using Ubuntu/Linux. 
3.  I used the wget command to get the ldc from GitHub. 
4.  I used the tar command to extract the ldc tar file. 
5. I updated the Path and Library Path to where the bin and library are located on my computer. 
export PATH=$PATH:/path/to/ldc/bin
export LIBRARY_PATH=$LIBRARY_PATH:/path/to/ldc/lib
6. I typed the command sambamba and it says the command is not recognized. 

What did I do wrong?

**Comment by pjotrp** [comment]
Please read https://github.com/biod/sambamba#getting-help

I suggest trying the latest prerelease binary https://github.com/biod/sambamba#binary-installation

**Comment by AndreaHobby** [comment]
I installed the binary file. 
I got through these steps. 
wget tarball.tar.bz2
mkdir ~/tmp
tar xvjf tarball.tar.bz2
./install.sh ~/sambamba-test

Now, I am trying to complete this step. However, it says there is no such file or directory. I looked and I see no bin folder inside of my  sambamba-test test folder. Where can I find the bin folder that is supposed to be here?
~/sambamba-test/bin/sambamba
